initfs/scripts/moslo-build.sh

```diff
@@ -60,17 +60,43 @@ debug()
 
 LIB_PATHS="/usr/lib /lib"
 
-# Function to find _direct_ lib dependencies for executable using objdump
-objdump_find_lib_path()
+# Recursive function to find all lib dependencies for a binary using objdump
+# $1 path of library / executable to search deps for
+# $2 temporary search file to store intermediate results, rm -f after call.
+objdump_find_lib_paths()
 {
-        local FOUND_LIBS=""
+        local RECURSED_LIBS=""
         local LIB_LIST=""
         local FOUND_LIB=""
+        local FOUND_LIBS=""
 
         if test -z $1; then
-                echo "find_lib_path: Error: Empty executable name!"
+                echo "$0: Error: Empty executable name!" 1>&2
+                exit 1
+        fi
+
+        if ! test -f $2; then
+                echo "$0: Error: No temporary list file provided!" 1>&2
                 exit 1
         fi
+
+        if cat $2 | grep -w $1; then
```

Comment:
Hah :)

I've been complained about my cat|grepping before, but this is my personal fixation stemming from 1st university course about usage of Unix. "in unix we have this thing called pipes.. so this is how you search something from a file cat | grep...". I think Janne Lääkkö (the original author of mosle-build) is also from Oulu university (the cat | sort | uniq copy-paste) ;)